### PR TITLE
SAK-42423 Google Drive integration with filepicker

### PR DESCRIPTION
**Pull request by bgarciaentornos**

### CODE REVIEW

File: cloud-storage/googledrive/api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+        </dependency>
+        <!--<dependency>
+            <groupId>com.google.api-client</groupId>
+            <artifactId>google-api-client</artifactId>
+            <version>1.30.2</version>
+        </dependency>-->
```
Comment (adrianfish) [inline cloud-storage/googledrive/api/pom.xml]: Why's this commented out?

File: cloud-storage/googledrive/api/src/java/org/sakaiproject/googledrive/model/GoogleDriveItem.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    private String googleDriveItemId;
+    private String name;    
+    private Long size;
+    private String downloadUrl;
+    private String viewUrl;
+    private String mimeType;
+    private String parentId;
+	private String icon;
+	private String thumbnail;
```
Comment (adrianfish) [inline cloud-storage/googledrive/api/src/java/org/sakaiproject/googledrive/model/GoogleDriveItem.java]: indentation

File: cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+		return false;
+	}
+
+	public Drive refreshToken(String userId){
+		if(!isConfigured()){
+			return null;
+		}
+		Drive googledriveUser = googledriveUserCache.get(userId);
+		if(googledriveUser != null) {
+			log.debug("refreshToken : Reusing previous user data " + googledriveUser);
```
Comment (adrianfish) [inline cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java]: You should probably guard this debug call.

File: cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+			Drive.Files.List list = service.files().list();
+			list.setFields(String.format("nextPageToken, files(%s)", String.join(",", fileFieldsToRequest)));
+			list.setOrderBy("folder,name");
+			String queryString = "'root' in parents";
+			list.setQ(queryString);
+			FileList result = list.execute();
+			List<File> files = result.getFiles();
+			for (File file : files) {
+				log.debug("Google data : {}", file);
```
Comment (adrianfish) [inline cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java]: You should guard this.

File: cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java (hunk 3 above)
Comment (bjones86) [reply cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java]: Guard is only necessary when using string concatenation, like is happening here. If you use the new logger style with replacement values, guarding is not necessary as the string concat doesn't happen regardless of logger level. In a nutshell, guard this here or convert it to the new style.

File: cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java (hunk 4 above)
Comment (bjones86) [reply cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java]: And this one doesn't need to be guarded.

File: cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java (hunk 3 above)
Comment (ottenhoff) [reply cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java]: New style much cleaner:

log.debug("refreshToken : Reusing previous user data {}", googledriveUser);

File: content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm [hunk 5]
```diff
@@ -624,6 +639,85 @@
...
+													#elseif($googledriveItem.hasChildren() && $googledriveItem.isExpanded())
+														#set($folderClass = "fa fa-folder-open")
+													#end
+													<a	#if($googledriveItem.hasChildren() && !$googledriveItem.isExpanded()) href="#toolLinkParam("FilePickerAction" "doNavigate" "googledriveCollectionId=${validator.escapeUrl(${googledriveItem.googleDriveItemId})}&googledriveCollectionDepth=${googledriveItem.depth}")" title="$tlang.getString('sh.go')" #end
+													><span class="$folderClass" aria-hidden="true"></span>
+													<span class="sr-only">$tlang.getString('gen.folder1')</span>
+												#else
+													<a target="_blank" class="googlename" href="$googledriveItem.viewUrl">
+														#if($googledriveItem.icon)
+															<img src="$googledriveItem.icon"/>
```
[the remark this answers is not in the document]
Comment (bjones86) [reply content/content-tool/tool/src/webapp/vm/content/sakai_filepicker_attach.vm]: Looks like this is still outstanding, but all the other Codacy complains were invalid so I resolved them.

File: cloud-storage/googledrive/api/pom.xml (hunk 1 above)
Comment (bgarciaentornos) [reply cloud-storage/googledrive/api/pom.xml]: I needed it at some point but not in the final version, removed it.

**Pull request merged**